### PR TITLE
Update to the beta version

### PR DESCRIPTION
**Pull request by michaelarick**
Update to the beta version to allow the embed to be via a div and JS code
not via an iframe. This version inserts an iframe in divs with the correct
class and data attribute.

### CODE REVIEW

File: change-embeds.js [hunk 1]
```diff
@@ -1,29 +1,22 @@
...
+      //Make sure the div hasn't already received its iframe
+      if (div.getAttribute('data-iframe-loaded') === 'true') { return; }
+      div.innerHTML = "<iframe height='530px' width='300px' src='https://www.change.org/embed/p/" +
+        petitionId + "/preview' frameborder='0'></iframe>";
+      div.setAttribute('data-iframe-loaded', 'true');
     }
   }
+}();
+
+document.addEventListener("DOMContentLoaded", ChangeEmbeds.insertIframesInDivs);
```
Comment (mattpardee) [inline change-embeds.js:24]: what browsers is this supported in?

File: change-embeds.js [hunk 2]
```diff
@@ -1,29 +1,22 @@
...
-        newWidth = midWidth;
-      } else {
-        newHeight = maxHeight;
-        newWidth = maxWidth;
+var ChangeEmbeds = function() {
+  return {
+    insertIframesInDivs: function() {
+      var divs = document.getElementsByClassName('change-embed-petition'),
+        i;
+      for (i = 0; i < divs.length; i ++) {
```
Comment (mattpardee) [inline change-embeds.js]: Slightly more efficient:

``` javascript
for (i = 0, len = divs.length; i < len; i++) {

}
```

File: change-embeds.js [hunk 3]
```diff
@@ -1,29 +1,22 @@
...
       }
-      div.style.height = newHeight.toString().concat('px');
-      iframe.style.height = newHeight.toString().concat('px');
-      iframe.style.width = newWidth.toString().concat('px');
+    },
+
+    insertIframeIntoDiv: function(div) {
+      var petitionId = div.getAttribute('data-petition-id');
+      //Make sure the div hasn't already received its iframe
+      if (div.getAttribute('data-iframe-loaded') === 'true') { return; }
```
Comment (mattpardee) [inline change-embeds.js]: and this works in all browsers?

**Comment by mattpardee** [comment]
Cool. It'd be great to have a minifier on this at some point, but obviously not a blocker (nor relevant to this PR)

File: change-embeds.js (hunk 1 above)
Comment (michaelarick) [inline change-embeds.js:24]: IE9+ and everybody else.
http://caniuse.com/#search=DOMContentLoaded

File: change-embeds.js (hunk 2 above)
Comment (michaelarick) [inline change-embeds.js]: ok.

File: change-embeds.js (hunk 3 above)
Comment (michaelarick) [inline change-embeds.js]: Supposedly: http://caniuse.com/#search=getAttribute

File: change-embeds.js [hunk 4]
```diff
@@ -1,29 +1,22 @@
...
-        newWidth = midWidth;
-      } else {
-        newHeight = maxHeight;
-        newWidth = maxWidth;
+var ChangeEmbeds = function() {
+  return {
+    insertIframesInDivs: function() {
+      var divs = document.getElementsByClassName('change-embed-petition'),
+        i, len;
+      for (i = 0, len = divs.length; i < len; i ++) {
```
Comment (mattpardee) [inline change-embeds.js]: nit: `i++` not `i ++`

above, the declaration of `len` should be on a new line

File: change-embeds.js (hunk 3 above)
Comment (mattpardee) [inline change-embeds.js]: :+1:

File: change-embeds.js (hunk 1 above)
Comment (mattpardee) [inline change-embeds.js:24]: :+1:

File: change-embeds.js [hunk 5]
```diff
@@ -1,29 +1,24 @@
...
-      div.style.height = newHeight.toString().concat('px');
-      iframe.style.height = newHeight.toString().concat('px');
-      iframe.style.width = newWidth.toString().concat('px');
+    },
+
+    insertIframeIntoDiv: function(div) {
+      var petitionId = div.getAttribute('data-petition-id');
+      if (!petitionId) {return;}
+      //Make sure the div hasn't already received its iframe
+      if (div.getAttribute('data-iframe-loaded') === 'true') {return;}
```
Comment (mattpardee) [inline change-embeds.js:14]: this should go at the top of the method

**Pull request merged**